From 32eb8357bb164b1b0d4f4a2df16dc2eada645b87 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Tue, 2 Jan 2018 09:03:45 +0000
Subject: [PATCH] Allow search all pid dirs when managing net_conf_t files

Port patch from https://github.com/fedora-selinux/selinux-policy:
  72511754 Allow search all pid dirs when managing net_conf_t files.

Fix avc denial:
  avc: denied { search } for pid=661 comm="dhclient-script" \
  name="resolve" dev="tmpfs" ino=16673 \
  scontext=system_u:system_r:dhcpc_t \
  tcontext=system_u:object_r:systemd_resolved_var_run_t tclass=dir

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/sysnetwork.if | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/policy/modules/system/sysnetwork.if b/policy/modules/system/sysnetwork.if
index 78ea93f..d23573a 100644
--- a/policy/modules/system/sysnetwork.if
+++ b/policy/modules/system/sysnetwork.if
@@ -355,6 +355,7 @@ interface(`sysnet_read_config',`
 	')
 
 	ifdef(`distro_redhat',`
+		files_search_all_pids($1)
 		allow $1 net_conf_t:dir list_dir_perms;
 		read_files_pattern($1, net_conf_t, net_conf_t)
 	')
@@ -487,6 +488,7 @@ interface(`sysnet_manage_config',`
 	')
 
 	ifdef(`distro_redhat',`
+		files_search_all_pids($1)
 		manage_files_pattern($1, net_conf_t, net_conf_t)
 	')
 ')
-- 
2.7.4

